### PR TITLE
test: improve gherkin suite with spec information and testdata

### PR DESCRIPTION
**Pull request by aepfli**
Adding newer evaluation tests, in the usual style. Additionally i am adding a json file with test data, so we do not need to manually create it all the time, but could use a json parser

### CODE REVIEW

File: specification/assets/gherkin/test-flags.json [hunk 1]
```diff
@@ -0,0 +1,186 @@
+{
+  "boolean-flag": {
+    "variants": {
+      "on": true,
+      "off": false
+    },
+    "defaultVariant": "on",
+    "flagMetadata": null
+  },
+  "boolean-targeted-zero-flag": {
```
Comment (alexandraoberaigner) [inline specification/assets/gherkin/test-flags.json:19]: [QUESTION] wondering if the naming could be improved here: `context-aware` & `*-targeted-*` seem to have similar test cases, but the naming suggests otherwise - those terms are so overloaded that I'd suggest to stick to one

Comment (aepfli) [reply specification/assets/gherkin/test-flags.json:19]: renamed flag, and added a bigger rule to the context-aware flag - we want to test more complicated targeting rules, and ensure that the fields are provided.

File: specification/assets/gherkin/evaluation_v2.feature [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+            | string-disabled-flag  | String  | bye     |
+
+  # Spec 1.4.13: Testing error message field in abnormal execution
+  # Testing: evaluation details may contain error message with additional details
+    @error-handling @spec-1.4.13
+    Scenario: Error message in evaluation details
+        Given a Boolean-flag with key "error-flag" and a default value "false"
+        When the flag was evaluated with details
+        Then the reason should be "ERROR"
+        And the error message should contain "flag error-flag not found"
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]: Maybe we should rename this to missing-flag

File: specification/assets/gherkin/evaluation_v2.feature [hunk 3]
```diff
@@ -0,0 +1,580 @@
...
+        @numbers
+        Examples: Number evaluations
+            | key               | type    | default | resolved_value |
+            | integer-zero-flag | Integer | 1       | 0              |
+            | float-zero-flag   | Float   | 0.1     | 0.0            |
+
+        @objects
+        Examples: Object evaluations
+            | key         | type   | default | resolved_value                                                                     |
+            | object-flag | Object | {}      | {\"showImages\": true,\"title\": \"Check out these pics!\",\"imagesPerPage\": 100} |
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]: This one is also correct

File: specification/assets/gherkin/evaluation_v2.feature [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+        And the evaluation details should be immutable
+
+  # Testing async/non-blocking mechanisms (Spec 1.4.12)
+  # Testing: client should provide asynchronous mechanisms for flag evaluation
+    @async @spec-1.4.12
+    Scenario: Asynchronous flag evaluation
+        Given a Boolean-flag with key "async-flag" and a default value "false"
+        When the flag was evaluated with details asynchronously
+        Then the evaluation should complete without blocking
+        And the result should be equivalent to synchronous evaluation
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]: I think I will change this to the other flags, the async flag does not differ in characteristics

File: specification/assets/gherkin/evaluation_v2.feature [hunk 5]
```diff
@@ -0,0 +1,580 @@
...
+    @reason-codes @spec-1.4.7
+    Scenario Outline: DISABLED reason
+        Given a <type>-flag with key "<key>" and a default value "<default>"
+        When the flag was evaluated with details
+        Then the resolved details value should be "<default>"
+        And the reason should be "DISABLED"
+        Examples:
+            | key                   | type    | default |
+            | boolean-disabled-flag | Boolean | false   |
+            | string-disabled-flag  | String  | bye     |
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]: In memory does not support state currently. Not sure if we should add state or not.

File: specification/assets/gherkin/evaluation_v2.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]: Furthermore the question is, if we do need those steps, as they are pretty provider specific. Maybe we should not test this for the SDK, but keep them for the providers

File: specification/assets/gherkin/evaluation_v2.feature (hunk 4 above)
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]: Furthermore this test will be most likely not implemented in the SDK. This is a provider implementation detail

File: specification/assets/gherkin/evaluation_v2.feature [hunk 6]
```diff
@@ -0,0 +1,617 @@
...
+            | object-disabled-flag | Object | {\"a\": 1} |
+
+  # Spec 1.4.13: Testing error message field in abnormal execution
+  # Testing: evaluation details may contain error message with additional details
+    @error-handling @spec-1.4.13
+    Scenario: Error message in evaluation details
+        Given a Boolean-flag with key "missing-flag" and a default value "false"
+        When the flag was evaluated with details
+        Then the reason should be "ERROR"
+        And the error message should contain "flag error-flag not found"
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]:
```suggestion
        And the error message should contain "flag missing-flag not found"
```

File: specification/assets/gherkin/evaluation_v2.feature [hunk 7]
```diff
@@ -0,0 +1,617 @@
...
+        @numbers
+        Examples: Number evaluations
+            | key               | type    | default | resolved_value |
+            | integer-zero-flag | Integer | 1       | 0              |
+            | float-zero-flag   | Float   | 0.1     | 0.0            |
+
+        @objects
+        Examples: Object evaluations
+            | key              | type   | default | resolved_value                                                                     |
+            | object-zero-flag | Object | {}      | {\"showImages\": true,\"title\": \"Check out these pics!\",\"imagesPerPage\": 100} |
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]:
```suggestion
            | object-zero-flag | Object | {\"a\": 1} | {}                                                                                 |
```

File: specification/assets/gherkin/evaluation_v2.feature [hunk 8]
```diff
@@ -0,0 +1,617 @@
...
+
+  # Spec 1.4.7: Testing DEFAULT reason when targeting doesn't match
+  # Testing: fallback to default value when context doesn't match targeting rules
+    @targeting @spec-1.4.7
+    Scenario Outline: Resolves zero value with targeting using default
+        Given a <type>-flag with key "<key>" and a default value "<default>"
+        And a context containing a key "email", with type "String" and with value "ballmer@none.com"
+        When the flag was evaluated with details
+        Then the resolved details value should be "<resolved_value>"
+        And the reason should be "DEFAULT"
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature:104]: i think claude is miss interpreting default with the default value, maybe we should call it fallback in the steps

File: specification/assets/gherkin/README.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  "email": "test@example.com",
+  "customer": true,
+  "age": 30
+}
+```
+
+## Contributing
+
+When modifying test data:
+1. Maintain the three-category structure (standard, zero, targeted-zero)
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/README.md]:
```suggestion
1. Maintain the category structure (standard, zero, targeted-zero, disabled)
```

File: specification/assets/gherkin/test-flags.json [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+    },
+    "contextEvaluator": "email == 'jobs@orange.com' ? 'special' : ''"
+  },
+  "no-default-flag-undefined-targeting-variant": {
+    "variants": {
+      "normal": "CFO",
+      "special": "CEO"
+    },
+    "contextEvaluator": "email == 'jobs@orange.com' ? 'special' : ''"
+  },
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/test-flags.json]: i think we need to evaluate this at a later stage anyways... not sure if we need those flags at all

File: specification/assets/gherkin/evaluation_v2.feature [hunk 11]
```diff
@@ -0,0 +1,617 @@
+Feature: Flag Evaluations - Complete OpenFeature Specification Coverage
+  # This comprehensive test suite covers all OpenFeature specification requirements
+  # for flag evaluation, including error handling, provider states, and edge cases.
+  # It reuses existing step definitions where possible and adds new scenarios for full compliance.
```
Comment (chrfwow) [inline specification/assets/gherkin/evaluation_v2.feature]: I don't think this line is relevant

Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]:
```suggestion
```

File: specification/assets/gherkin/test-flags.json (hunk 10 above)
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/test-flags.json]:
```suggestion
```

File: specification/assets/gherkin/evaluation_v2.feature [hunk 12]
```diff
@@ -0,0 +1,603 @@
...
+        @objects
+        Examples: Object evaluations
+            | key                       | type   | default    | resolved_value |
+            | object-targeted-zero-flag | Object | {\"a\": 1} | {}             |
+
+  # Spec 1.4.7: Testing DEFAULT reason when targeting doesn't match
+  # Testing: fallback to default value when context doesn't match targeting rules
+    @targeting @spec-1.4.7
+    Scenario Outline: Resolves zero value with targeting using default
+        Given a <type>-flag with key "<key>" and a default value "<default>"
```
Comment (toddbaert) [inline specification/assets/gherkin/evaluation_v2.feature]:
```suggestion
        Given a <type>-flag with key "<key>" and a fallback value "<default>"
```

For the test suite, lets try to use "fallback" for these cases to reduce confusion.

File: specification/assets/gherkin/evaluation_v2.feature [hunk 13]
```diff
@@ -0,0 +1,603 @@
...
+        @objects
+        Examples: Object evaluations
+            | type   | default    |
+            | Object | {\"a\": 1} |
+
+  # Spec 1.3.4: Testing type mismatch handling - wrong type evaluation
+  # Testing: client should return default value when provider returns wrong type
+  # Note: Test data setup ensures the flag returns wrong type, triggering type mismatch
+  # Implicitly tests: 1.4.10 (client never throws exceptions - returns default instead)
+    @error-handling @spec-1.3.4 @spec-1.4.10
```
Comment (toddbaert) [inline specification/assets/gherkin/evaluation_v2.feature]:
```suggestion
    @error-handling @error-handling-types @spec-1.3.4 @spec-1.4.10
```

I don't know if this is exactly the best term, but I think it's valuable to be able to add the granularity to be able to select on different sorts of errors, since, for example, some providers will do type coercion or don't even have typed APIs.

File: specification/assets/gherkin/evaluation_v2.feature [hunk 14]
```diff
@@ -0,0 +1,603 @@
...
+            | float-disabled-flag   | Float   | 0.1     |
+
+        @objects
+        Examples: Object evaluations
+            | key                  | type   | default    |
+            | object-disabled-flag | Object | {\"a\": 1} |
+
+  # Spec 1.4.13: Testing error message field in abnormal execution
+  # Testing: evaluation details may contain error message with additional details
+    @error-handling @spec-1.4.13
```
Comment (toddbaert) [inline specification/assets/gherkin/evaluation_v2.feature]: Same as https://github.com/open-feature/spec/pull/336/files#r2315433976

File: specification/assets/gherkin/evaluation_v2.feature [hunk 15]
```diff
@@ -0,0 +1,603 @@
...
+
+        @objects
+        Examples: Object evaluations
+            | key                       | type   | default    | resolved_value |
+            | object-targeted-zero-flag | Object | {\"a\": 1} | {}             |
+
+  # Spec 1.4.8 & 1.4.9: Testing FLAG_NOT_FOUND error code in abnormal execution
+  # Testing: client must return default value and error code when flag doesn't exist
+  # Implicitly tests: 1.4.10 (client never throws exceptions - returns default instead)
+    @error-handling @spec-1.4.8 @spec-1.4.9 @spec-1.4.10
```
Comment (toddbaert) [inline specification/assets/gherkin/evaluation_v2.feature]: Same as https://github.com/open-feature/spec/pull/336/files#r2315433976

File: specification/assets/gherkin/evaluation_v2.feature [hunk 16]
```diff
@@ -0,0 +1,603 @@
...
+
+        @objects
+        Examples: Object evaluations
+            | key         | type   | default    |
+            | object-flag | Object | {\"a\": 1} |
+
+  # Spec 1.7.7: Testing PROVIDER_FATAL error when provider is in fatal state
+  # Testing: client must short-circuit and return error when provider is fatal
+  # Implicitly tests: 1.4.10 (client never throws exceptions - returns default instead), 1.7.5 (FATAL status)
+    @provider-status @spec-1.7.7 @spec-1.4.10 @spec-1.7.5
```
Comment (toddbaert) [inline specification/assets/gherkin/evaluation_v2.feature]:
```suggestion
    @provider-status @provider-status-fatal @spec-1.7.7 @spec-1.4.10 @spec-1.7.5
```

Similar to https://github.com/open-feature/spec/pull/336/files#r2315433976, I think... but use your judgement.

File: specification/assets/gherkin/evaluation_v2.feature [hunk 17]
```diff
@@ -0,0 +1,603 @@
...
+            | key          | type    | default | resolved_value |
+            | integer-flag | Integer | 1       | 10             |
+            | float-flag   | Float   | 0.1     | 0.5            |
+            
+        @objects
+        Examples: Object evaluations
+            | key         | type   | default | resolved_value                                                                     |
+            | object-flag | Object | {}      | {\"showImages\": true,\"title\": \"Check out these pics!\",\"imagesPerPage\": 100} |
+#
+# IMPLICIT SPECIFICATION COVERAGE NOTES:
```
Comment (toddbaert) [inline specification/assets/gherkin/evaluation_v2.feature:531]: :+1:

**Review by toddbaert — APPROVED** [review]

File: specification/assets/gherkin/evaluation_v2.feature [hunk 18]
```diff
@@ -0,0 +1,587 @@
...
+        Examples: Number evaluations
+            | key          | type    | default | expected_variant |
+            | integer-flag | Integer | 1       | ten              |
+
+            | float-flag   | Float   | 0.1     | half             |
+
+        @objects
+        Examples: Object evaluations
+            | key         | type   | default | expected_variant |
+            | object-flag | Object | {}      | template   |
```
[the remark this answers is not in the document]
Comment (aepfli) [reply specification/assets/gherkin/evaluation_v2.feature]:
```suggestion
            | object-flag | Object | {}      | template         |
```

File: specification/assets/gherkin/evaluation.feature [hunk 19]
```diff
@@ -1,3 +1,4 @@
+@deprecated
```
Comment (askpt) [inline specification/assets/gherkin/evaluation.feature:1]: What is the consequences of this tag for the consumers of the `.feature`? Tried to investigate but I couldn't find any.

**Review by askpt — APPROVED** [review]
LGTM. Just a minor comment out of curiosity.

File: specification/assets/gherkin/evaluation.feature (hunk 19 above)
Comment (aepfli) [reply specification/assets/gherkin/evaluation.feature:1]: The new gherkin implementations should autoload all gherkin files in this directory - with @deprecated we add the possibility to filter this test out, as we might need it for sdks not migrated to the new way of testing. - it is a gherkin test filter
